App.js

```diff
@@ -28,10 +28,28 @@ const RootStack = createSwitchNavigator(
 )
 
 export default class App extends Component {
+
+  constructor() {
+    super()
+    this.state = {
+      userId: ''
+    }
+
+    this._bootstrapAsync()
+  }
+
+  _bootstrapAsync = async () => {
+    this.setState({
+      userId: await AsyncStorage.getItem('userId')
```

Comment:
I like the idea of reading AsyncStorage here and then using it from the _state_, seems cleaner to me :+1:

However, shouldn't we collect the user data from here and behave accordingly?

```
const userId = await AsyncStorage.getItem('userId')
const user = _call the api here_
if (!user) _go to login page_
this.setState({ user })

Reply:
yes sure, it's my way 🥇